docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This integer property accepts values 1 to 10 inclusive.")
+  @RangeInt(min = 1, max = 10)
+  public static int myIntegerProperty = false;
+
+  @Comment("This double property accepts values 0.0 to 1.0 inclusive.")
+  @RangeDouble(min = 0.0, max = 1.0)
+  public static double myDoubleProperty = false;
+
+}
+````
+
+!! important
+
+    While both the `@RangeInt` and `@RangeDouble` accept a min and a max value to be set they are not interchangeable, given that the type must match. Consequently the `@RangeInt` annotation is only to be applied to `int` and `Integer` fields, and the `@RangeDouble` is only to be applied to `double` and `Double` fields.
+
+!! important
+
+    There currently exists no range annotations capable of supporting numeric types other than the aforementioned `int`, `Integer`, `double`, and `Double` types. As such, you are forced to use `int` instead of `byte` or `long`, `Integer` instead of `Byte` or `Long`, `double` instead of `float`, and `Double` instead of `Float`. This issue will likely be fixed in a future release given wide enough adoption of the new annotation-based configuration system.
+
+
+Sub-categories
+-------------------------
+
+Smaller mods will generally have fewer properties (say about 5 - 10) and do not need to worry about dividing up their configurations into more manageable groups of properties for the end-user to manage. However, for larger mods configurations consisting of 20 to 50 elements is not all that uncommon, and the need for sub-categories becomes irrefutable. Recognizing such a need, the annotation-based configuration system provides a mechanism by which `@Config` classes can be divided into groups that are reflected directly in the structure of the `@Config` class.
+
+The sub-division of `@Config` classes is accomplished by the addition of `public static` inner classes that act as the physical configuration groupings. Thus, any property that is placed in the inner class is then resolved by the system as not to exist in the default group defined by the `@Config` annotation but rather a new group with the name of the inner class.
+
+Building upon the prior examples, an example usage of property grouping is as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property belongs to the default group.")
+  public static boolean myPropertyDefaultGroup = false;
+
+  public static class MyCategory {
+
+    @Comment("This property belongs to the MyCategory group.")
+    public static boolean myPropertyMyCategory = false;
+
+  }
+
+}
+````
+
+
+Alternate Names
+-------------------------
+
+Sometimes, for reasons unknown, we wind up with variable names in a configuration class that would be unsuitable for use as an actual property key. This would be problematic as Forge determines the property key based off of variable names. For this reason, the `@Name` annotation exists.
```

Comment:
Too much fluff. I'd reword this to

>Using the `@Name` annotation, the property can have a key or name specified that is different from its field's name.

Really that's all the information that's needed here.